include/poggles/shader.h

```diff
@@ -12,6 +12,13 @@
 namespace poggles
 {
 
+struct shader_desc
+{
+  GLenum type;
+  std::string name;
```

Comment:
In general I like this struct approach, but I wonder if there is a better type for the name attribute. My first instinct is a `std::filesystem::path`, but that isn't quite right. Although it should ne netter than a string

Reply:
I'm with you here, but I can't think of a better type for a fit. It might be worth revisiting this later